OptimizelySDKCore/OptimizelySDKCore/Optimizely.m

```diff
@@ -880,18 +866,69 @@ - (OPTLYVariation *)sendImpressionEventFor:(OPTLYExperiment *)experiment
                                                  callback(error);
                                              }
                                          }];
-    NSString *_userId = userId ? userId : @"";
-    NSDictionary *_attributes = attributes ? attributes : [NSDictionary new];
-    [_notificationCenter sendNotifications:OPTLYNotificationTypeActivate
-                                      args:[NSArray arrayWithObjects:experiment,
-                                            _userId,
-                                            _attributes,
-                                            variation,
-                                            impressionEventParams,
-                                            nil]];
+    
+    NSArray *args = @[experiment, userId ? : @"", attributes ? : [NSDictionary new], variation, impressionEventParams];
+    [_notificationCenter sendNotifications:OPTLYNotificationTypeActivate args:args];
     return variation;
 }
 
+/**
+ Helper method to retrieve decisions for the user for experiment(s) using the provided event.
+
+ @param event The event which needs to be recorded.
+ @param userId Id for user.
+ @param attributes The user's attributes.
+ @return Array of dictionaries containing valid experiment Ids, variation Ids and layer Ids into which the user is bucketed.
+ */
+- (NSArray<NSDictionary *> *)decisionsFor:(OPTLYEvent *)event
```

Comment:
I didn't get the context of this method.

Reply:
Helper method to retrieve decisions for the user from the provided event.